exercises/react/example.cr

```diff
@@ -0,0 +1,76 @@
+class Cell
+  @dependents = [] of ComputeCell
+
+  def initialize(initial : Int32)
+    @value = initial
+  end
+
+  getter :value
+
+  protected getter :dependents
+end
+
+class InputCell < Cell
+  def value=(new_value : Int32)
+    @value = new_value
+    @dependents.each(&.propagate)
+    @dependents.each(&.fire_callbacks)
+  end
+end
+
+class ComputeCell < Cell
+  alias CallbackID = Int32
+
+  @callbacks_given = 0
+  @callbacks = {} of CallbackID => Int32 ->
+  @last_callback_value : Int32
+
+  def initialize(new_value : -> Int32, *inputs : Cell)
+    super(new_value.call)
+    @new_value = new_value
+    @last_callback_value = @value
+    inputs.each { |i| i.dependents << self }
+  end
+
+  def add_callback(&block : Int32 ->) : CallbackID
+    @callbacks_given += 1
+    @callbacks[@callbacks_given] = block
+    @callbacks_given
+  end
+
+  def remove_callback(id : CallbackID)
+    @callbacks.delete(id)
+  end
+
+  def propagate
```

Comment:
Never mind, problem solved for real this time, I think. Add `protected def` for these functions on `Cell`, then override it at `ComputeCell`. Works, and non-`Cell` now can't call these functions.